components/activity_log_modal/activity_log_modal.tsx

```diff
@@ -2,64 +2,71 @@
 // See LICENSE.txt for license information.
 
 import $ from 'jquery';
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
+import {Session} from 'mattermost-redux/types/sessions';
+import {ActionFunc} from 'mattermost-redux/types/actions';
+
 import ActivityLog from 'components/activity_log_modal/components/activity_log';
-import LoadingScreen from 'components/loading_screen';
 
-export default class ActivityLogModal extends React.PureComponent {
-    static propTypes = {
+export type Props = {
 
-        /**
-         * The current user id
-         */
-        currentUserId: PropTypes.string.isRequired,
+    /**
+ * The current user id
+ */
+    currentUserId: string;
 
-        /**
-         * Current user's sessions
-         */
-        sessions: PropTypes.oneOfType([
-            PropTypes.array,
-            PropTypes.object,
-        ]).isRequired,
+    /**
+     * Current user's sessions
+     */
+    sessions: Session[];
+
+    /**
+     * Current user's locale
+     */
+    locale: string;
+
+    /**
+     * Function that's called when user closes the modal
+     */
+    onHide: () => void;
+
+    actions: {
 
         /**
-         * Current user's locale
+         * Function to refresh sessions from server
          */
-        locale: PropTypes.string.isRequired,
+        getSessions: (userId: string) => ActionFunc;
 
         /**
-         * Function that's called when user closes the modal
+         * Function to revoke a particular session
          */
-        onHide: PropTypes.func.isRequired,
-        actions: PropTypes.shape({
-
-            /**
-             * Function to refresh sessions from server
-             */
-            getSessions: PropTypes.func.isRequired,
-
-            /**
-             * Function to revoke a particular session
-             */
-            revokeSession: PropTypes.func.isRequired,
-        }).isRequired,
+        revokeSession: (userId: string, sessionId: string) => Promise<{ data: boolean }>;
+    };
+}
+
+type State ={
```

Comment:
Minor linting problem here:
```suggestion
type State = {
```